src/parenthesis.ts

```diff
@@ -0,0 +1,89 @@
+// [VexFlow](https://vexflow.com) - Copyright (c) Mohit Muthanna 2010.
+// Author: Rodrigo Vilar
+// MIT License
+
+import { Glyph } from './glyph';
+import { Modifier, ModifierPosition } from './modifier';
+import { ModifierContextState } from './modifiercontext';
+import { Note } from './note';
+
+/** Parenthesis implements parenthesis modifiers for notes.*/
+export class Parenthesis extends Modifier {
+  static get CATEGORY(): string {
+    return 'Parenthesis';
+  }
+
+  protected scale: number;
+
+  // Arrange dots inside a ModifierContext.
+  static format(parentheses: Parenthesis[], state: ModifierContextState): boolean {
+    const { right_shift, left_shift } = state;
+
+    if (!parentheses || parentheses.length === 0) return false;
+
+    let x_widthL = 0;
+    let x_widthR = 0;
+
+    for (let i = 0; i < parentheses.length; ++i) {
+      const parenthesis = parentheses[i];
+      const note = parenthesis.getNote();
+      const pos = parenthesis.getPosition();
+      const index = parenthesis.checkIndex();
+      const props = note.getKeyProps()[index];
+
+      let shift = 0;
+
+      if (pos === ModifierPosition.RIGHT) {
+        shift = (props.displaced ? note.getRightDisplacedHeadPx() : 0) + right_shift;
+        x_widthR += shift + parenthesis.width;
+      }
+      if (pos === ModifierPosition.LEFT) {
+        shift = (props.displaced ? note.getLeftDisplacedHeadPx() : 0) + left_shift;
+        x_widthL += shift + parenthesis.width;
+      }
+      parenthesis.setXShift(shift);
+    }
+    state.left_shift += x_widthL;
+    state.right_shift += x_widthR;
+
+    return true;
+  }
+
```

Comment:
Add another static function `build(targetNote: Note)` and move the code from `stavenote.ts` here which builds both the left and right parenthesis.

Reply:
I have called it `addParentheses` rather than build. This is inline to the `getAccidentals` in the other PR. `Build` would also be misleading because to parenthesis are added rather than one parenthesis built.